src/main/java/mate/academy/MergeSortAction.java

```diff
@@ -1,8 +1,60 @@
 package mate.academy;
 
-public class MergeSortAction {
+import java.util.Arrays;
+import java.util.concurrent.RecursiveAction;
+
+public class MergeSortAction extends RecursiveAction {
+    private static final int THRESHOLD = 4;
+
+    private final int[] array;
+
+    private final int start;
+
+    private final int end;
 
     public MergeSortAction(int[] array) {
+        this(array, 0, array.length - 1);
+    }
+
+    public MergeSortAction(int[] array, int start, int end) {
+        this.array = array;
+        this.start = start;
+        this.end = end;
+    }
+
+    @Override
+    protected void compute() {
+        if (end - start + 1 <= THRESHOLD) {
```

Comment:
The condition 'end - start + 1 <= THRESHOLD' is correct for determining when to switch to a direct sort. However, ensure that the THRESHOLD value is appropriate for your use case, as a low threshold might not fully utilize the benefits of parallelism.